service/src/main/java/bio/terra/workspace/app/controller/shared/PropertiesUtils.java

```diff
@@ -28,4 +34,16 @@ public static ApiProperties convertMapToApiProperties(Map<String, String> proper
     properties.forEach((key, value) -> apiProperties.add(new ApiProperty().key(key).value(value)));
     return apiProperties;
   }
+
+  /**
+   * Clear certain properties in the hashmap before making a clone.
+   *
+   * <p>For example, TERRA_WORKSPACE_FOLDER_ID is a workspace specific properties. It needs to be
+   * cleared because it is meaningless in a new workspace.
+   */
+  public static ImmutableMap<String, String> clearSomeProperties(Map<String, String> properties) {
```

Comment:
UI question. Ignore Data Collection UI.

- Say in UI resource R is in folder F in workspace W
- Say I clone R into same workspace W
- By default, I would think the clone would be in the same folder F. Eg in google docs, if I make a copy, the copy is in the same folder.

I wonder if we should keep folder if source and dest workspace are the same.

@cbookg thoughts?

Reply:
good point. i think that makes sense to keep the folder properties if cloning into the same workspace.

Reply:
i modified it to keep folder if source and dest workspaceid are the same